content/en/docs/rules/fspath.md

```diff
@@ -0,0 +1,113 @@
+linktitle: Accessing File System Paths
+weight: 100
+---
+
+## Introduction
+
+This section explains how the fields `fs.path.*` work and when they can be used.
+
+### Motivation
+
+A variety of different syscalls take file system paths as arguments. However, there is little consistency in the fields that can access those file system paths. Depending on the syscall, the file system path might be available in `evt.rawarg.path`, `evt.rawarg.pathname`, `evt.rawarg.name`, `fd.name`, etc. And for some system calls, the file system path is actually in the enter event instead of the exit event. This makes it very difficult to write simple Falco rules that act on file system paths, because the field must depend on the syscall as well as the event direction.
+
+To help address this inconsistency, in Falco version 0.36 we added a new set of fields that normalize file system paths across a variety of syscalls.
+
+#### What Counts As A File System Path?
+
+Lots of existing fields also refer to file system paths. For example, `proc.exepath` contains the file system path for an executable, and there are related fields `proc.pexepath`/`proc.aexepath`. `container.mount.*` fields all contain the file system paths for file systems mounted into a container.
+
+The `fs.path.*` fields are *only* populated for syscalls that are generally related to reading, writing, or modifying some file system object and that have a file system path as an argument. The goal is to have a single set of fields that can be always relied on to refer to those paths, as compared to checking the widely varying per-event fields.
+
+### `fs.path.*` fields
+
+The following fields are available for any syscall that operates on a file system path:
+
+* fs.path.name
+* fs.path.nameraw
+* fs.path.source
+* fs.path.sourceraw
+* fs.path.target
+* fs.path.targetraw
+
+`fs.path.name` is for file operations that work on a path like open, unlink, rmdir, etc. For other file operations that have a source and target, like cp, symlink, link, mv, etc, there are fields `fs.path.source` and `fs.path.target`.
+
+All of these convert relative paths to absolute paths when needed, using the thread's current working directory (cwd).
+
+`fs.path.nameraw/fs.path.sourceraw/fs.path.targetraw` are like the above but do *not* convert relative paths to absolute paths. They always contain the original path, which may or may not be relative.
+
+The fields only work for exit events and only return a value if the syscall was successful.
+
+The below tables show:
+* the specific syscalls that are are supported
+* the specific falco event identifers are supported. The reason there are multiple event identifers for the same syscall (e.g. MKDIR vs MKDIR_2) is that libs used to define new events every time we added/modified arguments to the enter or exit event. Older applications using the older version of libs will use the older event identifier for the syscall name, while newer applications will use the newer event identifier.
```

Comment:
This is great it explains the difference between enter and exit events, and I am still not sure if we have to list the older event types in the table since we start introducing this with Falco 0.36. Going forward it makes sense to version this table. Alternatively it would even be ok to remove the column "Falco Libs Event Identifier" altogether as perhaps it confuses most Falco end users more than it helps. Falco adopters primarily think of the syscall string name as event I think and libs adopters likely are familiar with the source code and consult the event table. No real opinion here, just thinking out loud.

Reply:
We can remove it if it's not valuable. I think it's valuable to devs but not users.

Reply:
Maybe it's hard to keep that page updated with changing PPME codes? Let's remove it, we can rather add a link to the event_table source code?

Reply:
Ok I removed the PPME codes as well as the column that denoted whether the value comes from the enter or exit event.